File: tests/write_test.py
from time import sleep
from sys import exit
import os
FILE=open("PINCE-to-Inferior.txt", "w")
FILE.close()
FILE=open("Inferior-to-PINCE.txt", "w")
FILE.close()
FILE=open("status.txt", "w")
FILE.close()
while True:
	status_word="waiting"
	while status_word not in "sync-request-recieve":
		status=open("status.txt", "r")
		status_word=status.read()
		status.close()
		try:
			abort=open("abort.txt", "r")
			abort.close()
			os.remove("abort.txt")
			FILE=open("abort-verify.txt", "w")
			FILE.close()
			exit(0)
		except:
			pass
	status=open("status.txt", "w")
	status.write("sync-request-send")
	status.close()
	FILE=open("PINCE-to-Inferior.txt", "w")
	FILE.close()
	FILE=open("Inferior-to-PINCE.txt", "r")
	readed=FILE.read()
	print(readed)
	FILE.close()

File: tests/write_test.c
#include <stdio.h>
#include <unistd.h>

main() {
	FILE *fp;
	FILE *status;
	FILE *exit;
	FILE *initialize;
	char buff[255]="";
	char status_word[255]="";
	initialize = fopen("status.txt", "r");
	while(initialize==NULL){
		initialize = fopen("status.txt", "r");
	}
	fclose(initialize);
	while(1){
		sleep(1);
		status = fopen("status.txt", "w");
		fputs("sync-request-recieve", status);
		fclose(status);
		exit = fopen("abort-verify.txt", "r");
		if(exit!=NULL){
			fclose(exit);
			remove("abort-verify.txt");
			remove("status.txt");
			remove("PINCE-to-Inferior.txt");
			remove("Inferior-to-PINCE.txt");
			return 0;
		}
		fp = fopen("PINCE-to-Inferior.txt", "r");
		fscanf(fp, "%s", buff);
		printf("%s\n", buff );
		fclose(fp);
		fp = fopen("Inferior-to-PINCE.txt", "w");
		fputs("a", fp);
		fclose(fp);
	}
}
